File: apps/api/app/schemas/workflow.py
from datetime import datetime
from pydantic import BaseModel, Field


class WorkflowStepSchema(BaseModel):
    model_config = {"from_attributes": True}

    id: str
    workflow_profile_id: str
    stage: str
    provider_id: str | None
    model_id: str | None
    prompt_version_id: str | None
    temperature: float
    top_p: float
    max_output_tokens: int
    timeout_seconds: int


class WorkflowProfileSchema(BaseModel):
    model_config = {"from_attributes": True}

    id: str
    name: str
    description: str
    is_default: bool
    created_at: datetime
    updated_at: datetime
    steps: list[WorkflowStepSchema] = []


class WorkflowProfileListSchema(BaseModel):
    model_config = {"from_attributes": True}

    id: str
    name: str
    description: str
    is_default: bool
    created_at: datetime
    updated_at: datetime


class WorkflowCreate(BaseModel):
    name: str = Field(min_length=1, max_length=200)
    description: str = ""


class WorkflowUpdate(BaseModel):
    name: str | None = Field(default=None, max_length=200)
    description: str | None = None


class WorkflowStepUpdate(BaseModel):
    provider_id: str | None = None
    model_id: str | None = None
    prompt_version_id: str | None = None
    temperature: float | None = None
    top_p: float | None = None
    max_output_tokens: int | None = None
    timeout_seconds: int | None = None

File: apps/web/src/types/workflow.ts
export interface WorkflowStepConfig {
  id: string;
  workflow_profile_id: string;
  stage: string;
  provider_id?: string;
  model_id?: string;
  prompt_version_id?: string;
  temperature: number;
  top_p: number;
  max_output_tokens: number;
  timeout_seconds: number;
}

export interface WorkflowProfile {
  id: string;
  name: string;
  description: string;
  is_default: boolean;
  created_at: string;
  updated_at: string;
  steps: WorkflowStepConfig[];
}

export interface WorkflowProfileList {
  id: string;
  name: string;
  description: string;
  is_default: boolean;
  created_at: string;
  updated_at: string;
}

export interface WorkflowCreate {
  name: string;
  description?: string;
}

export interface WorkflowUpdate {
  name?: string;
  description?: string;
}

export interface WorkflowStepUpdate {
  provider_id?: string;
  model_id?: string;
  prompt_version_id?: string;
  temperature?: number;
  top_p?: number;
  max_output_tokens?: number;
  timeout_seconds?: number;
}
